src/main/webapp/app/course/manage/course-management.component.ts

```diff
@@ -53,6 +56,37 @@ export class CourseManagementComponent implements OnInit, OnDestroy, AfterViewIn
             (res: HttpResponse<Course[]>) => {
                 this.courses = res.body!;
                 this.courseForGuidedTour = this.guidedTourService.enableTourForCourseOverview(this.courses, tutorAssessmentTour, true);
+                this.courseSemesters = this.courses
+                    .map((c) => c.semester ?? '')
+                    .filter((value, index, self) => {
+                        // filter down to unique values
+                        return !this.courses[index].testCourse && self.indexOf(value) === index;
+                    })
+                    .sort((a, b) => {
```

Comment:
Maybe you can use here also `orderBy` from `lodash` see https://lodash.com/docs/4.17.15#orderBy

Reply:
And a bit more docu which magic you are doing would be good :D

Reply:
Added comments and renamed a few variables so it is hopefully more clear. Regarding `orderBy`: I don't know how to get that to work in this case because of our special cases (unset semester after all, and WS > SS when the years are equal).